libsolutil/Address.cpp

```diff
@@ -0,0 +1,73 @@
+/*
+	This file is part of solidity.
+
+	solidity is free software: you can redistribute it and/or modify
+	it under the terms of the GNU General Public License as published by
+	the Free Software Foundation, either version 3 of the License, or
+	(at your option) any later version.
+
+	solidity is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+	GNU General Public License for more details.
+
+	You should have received a copy of the GNU General Public License
+	along with solidity.  If not, see <http://www.gnu.org/licenses/>.
+*/
+// SPDX-License-Identifier: GPL-3.0
+
+#include <libsolutil/Address.h>
+#include <libsolutil/Exceptions.h>
+
+using namespace std;
+
+namespace solidity::util
+{
+
+/// @returns a h160 representation of an address string
+/// or an error message in case the address string is not a valid address
+Result<h160> validateAddress(string _addrString)
+{
+	string message;
+	h160 address;
+
+	if (_addrString.substr(0, 2) != "0x")
+		message = "Library address \"" + _addrString + "\" is not prefixed with \"0x\".\n"
+			"Note that the address must be prefixed with \"0x\".";
```

Comment:
Although I'm ok with the original style, I'm just wondering if all messages should be formatted as the one at line 42. I guess this was the idea initially suggested by @matheusaaguiar here https://github.com/ethereum/solidity/pull/14283#discussion_r1213526671, right?

```suggestion			
		message =
				"Library address \"" + 
				_addrString +
				"\" is not prefixed with \"0x\".\n"
				"Note that the address must be prefixed with \"0x\".";
```

Reply:
Well, I didn't mind with that one (and didn't point it specifically) because it's only the last part of the string literal and it is not really split or combined with an operator. For me it's like the second line here is just the first one "overlapping", if that makes sense.
For me it's fine either way.